fence/README.md

```diff
@@ -0,0 +1,155 @@
+perform node level fencing using corresponding CIDR blocks.
+
+### Goals in MVP
+
+The new extension will define a procedure that allows SP to blocklist
+any client using corresponding CIDR blocks.
+
+If multiple COs are accessing a shared resource provided by the SP, and a node
+failure takes place, this procedure will allow the SP to blocklist any node of CO,
+and thus preventing further access and corruption of data.
+
+### Non-Goals in MVP
+
+* Application level fencing.
+
+## Solution Overview
+
+This specification defines an interface along with the minimum operational and
+packaging recommendations for a storage provider (SP) to implement
+fencing operation. The interface declares the RPCs that a plugin
+MUST expose.
+
+## RPC Interface
+
+* **Controller Service**: The Controller plugin MUST implement these sets of
+  RPCs.
+
+```protobuf
+syntax = "proto3";
+package fence;
+
+import "google/protobuf/descriptor.proto";
+
+option go_package = ".;fence";
+
+extend google.protobuf.FieldOptions {
+  // Indicates that a field MAY contain information that is sensitive
+  // and MUST be treated as such (e.g. not logged).
+  bool fence_secret = 1099;
+}
+
+// Controller holds the RPC method for performing fencing operations.
+service Controller {
+  // FenceClusterNetwork RPC call to perform a fencing operation.
+  rpc FenceClusterNetwork (FenceClusterNetworkRequest)
+  returns (FenceClusterNetworkResponse) {}
+
+  // UnfenceClusterNetwork RPC call to remove a list of CIDR blocks from the
+  // list of blocklisted/fenced clients.
+  rpc UnfenceClusterNetwork (UnfenceClusterNetworkRequest)
+  returns (UnfenceClusterNetworkResponse) {}
+
+  // ListClusterFence RPC call to provide a list of blocklisted/fenced clients.
+  rpc ListClusterFence(ListClusterFenceRequest)
+  returns (ListClusterFenceResponse){}
+}
+```
+
+### FenceClusterNetwork
+
+```protobuf
+// FenceClusterNetworkRequest contains the information needed to identify
+// the cluster so that the appropriate fencing operation can be
+// performed.
+message FenceClusterNetworkRequest {
+  // Plugin specific parameters passed in as opaque key-value pairs.
+  map<string, string> parameters = 1;
+  // Secrets required by the plugin to complete the request.
+  map<string, string> secrets = 2 [(fence_secret) = true];
+  // list of CIDR blocks on which the fencing operation is expected to be
+  // performed.
+  repeated CIDR cidrs = 3;
+}
+
+// FenceClusterNetworkResponse holds the information about the result of the
+// FenceClusterNetworkRequest call.
+message FenceClusterNetworkResponse {
+}
+```
+
+### UnfenceClusterNetwork
+
+```protobuf
+// UnfenceClusterNetworkRequest contains the information needed to identify
+// the cluster so that the appropriate fence operation can be
+// disabled.
+message UnfenceClusterNetworkRequest {
+  // Plugin specific parameters passed in as opaque key-value pairs.
+  map<string, string> parameters = 2;
```

Comment:
why start with `2`?

Reply:
That must have been an overlook, updated now